components/UploadPopover/UploadPopover.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from "react";
+import styles from "./UploadPopover.module.css";
+import { Button, Popover, LinearProgress } from "@mui/material";
+import Icon from "../../assets/Icon";
+import { Transaction, User } from "../../types/schema";
+import { addTransaction } from "../../firebase/firestore/transaction";
+import Papa from '../../node_modules/papaparse';
+
+type UploadPopoverProps = {
+    uploadAnchor: Element;
+    closeUpload: Function;
+    popoverid: string;
+};
+
+export const UploadPopover: React.FunctionComponent<UploadPopoverProps> = ({
+  uploadAnchor,
+  closeUpload,
+  popoverid,
+}: UploadPopoverProps) => {
+    const [uploadFile, setUploadFile] = useState(null);
+    const [fileData, setFileData] = useState([]);
+    const [uploading, setUploading] = useState(false);
+    const [uploadProgress, setUploadProgress] = useState(0);
+    const [uploadSuccess, setUploadSuccess] = useState(false);
+
+    const sleep = (ms) => {
+        return new Promise(resolve => setTimeout(resolve, ms));
+    }
+
+    const handleUploadClose = () => {
+        closeUpload();
+        sleep(1000).then(() => {
+            setUploadSuccess(false);
+            setUploadFile(null);
+            setUploading(false);
+            setUploadProgress(0);
+        })
+    }
+
+    const handleUpload = (event) => {
+        console.log(event.target.files[0]);
+        setUploadFile(event.target.files[0]);
+    }
+
+    const cancelFile = () => {
+        setUploadFile(null);
+    }
+
+    const cancelUploading = () => {
+        setUploading(false);
+        setUploadProgress(0);
+    }
+
+    const handleUploadConfirm = () => {
+        if (uploading) {
+            //handle file uploads from fileData
+            for (let i = 1; i < fileData.length; i++) {
```

Comment:
in your next PR, could you add validation/error states for improperly formatted CSVs?